tests/test-tag.cpp

```diff
@@ -0,0 +1,22 @@
+#include <catch2/catch_test_macros.hpp>
+
+#include <log_surgeon/finite_automata/Tag.hpp>
+
+using log_surgeon::finite_automata::Tag;
+
+TEST_CASE("Tag operations", "[Tag]") {
+    SECTION("Basic name retrieval works correctly") {
+        Tag const tag("uID");
```

Comment:
```suggestion
        Tag const tag{"uID"};
```
Also for the rest two

Reply:
Whoops, copied rabbit's recommendation for the additional test cases and missed the lack of braced initialization.
Changed.